tools/pylib/boututils/calculus.py

```diff
@@ -105,24 +96,25 @@ def deriv2D(data,axis=-1,dx=1.0,noise_suppression=True):
   s = data.shape
   if axis > len(s)-1:
     raise RuntimeError("ERROR: axis out of bounds for derivative")
- 
-  if noise_suppression:  
+
+  if noise_suppression:
     if s[axis] < 11:
       raise RuntimeError("Data too small to use 11th order method")
     tmp = array([old_div(-1.0,512.0),old_div(-8.0,512.0),old_div(-27.0,512.0),old_div(-48.0,512.0),old_div(-42.0,512.0),0.0,old_div(42.0,512.0),old_div(48.0,512.0),old_div(27.0,512.0),old_div(8.0,512.0),old_div(1.0,512.0)])
   else:
     if s[axis] < 9:
       raise RuntimeError("Data too small to use 9th order method")
-    tmp = array([old_div(1.0,280.0),old_div(-4.0,105.0),old_div(1.0,5.0),old_div(-4.0,5.0),0.0,old_div(4.0,5.0),old_div(-1.0,5.0),old_div(4.0,105.0),old_div(-1.0,280.0)])       
-  
+    tmp = array([old_div(1.0,280.0),old_div(-4.0,105.0),old_div(1.0,5.0),old_div(-4.0,5.0),0.0,old_div(4.0,5.0),old_div(-1.0,5.0),old_div(4.0,105.0),old_div(-1.0,280.0)])
+
+  from scipy.signal import convolve
```

Comment:
I think it makes sense to move this import into the function, since most functions here do not depend on it. I think however that it should be at the start of the function to make the dependency clearer. 

PEP8 recommends putting imports at the top of the file, but there is some discussion of situations when that is not optimal (and I suspect this is one of them): https://stackoverflow.com/questions/128478/should-import-statements-always-be-at-the-top-of-a-module

Reply:
Yes, I thought so. I often enough don't have scipy installed - in which case I can still use the other functions.